core/primitives/src/congestion_info.rs

```diff
@@ -112,8 +113,47 @@ impl CongestionControl {
 
     /// Whether we can accept new transaction with the receiver set to this shard.
     pub fn shard_accepts_transactions(&self) -> bool {
-        self.congestion_level() < self.config.reject_tx_congestion_threshold
+        self.rejection_reason().is_none()
     }
+
+    /// If the shard doesn't accept new transaction, provide the reason for
+    /// extra debugging information.
+    pub fn rejection_reason(&self) -> Option<RejectTransactionReason> {
```

Comment:
mini nit: I think it's easier to work with positive statements in method names because in order to get a positive you need double negation like "not rejected". How about returning an enum like:
```
enum ShardAcceptsTransactions {
  Yes,
  No(Reason) // or inline reason to NoDueToIncomingCongestion, NoDueToMissedChunks, etc
}
```

totally optional as it's just my preference :)

Reply:
You are right, it would be better. I was a bit too lazy to do it right yesterday but gave it another spin now. Please let me know what you think of the latest refactor. I find it a bit more verbose but I think it resolves the double negation that you don't like.